### PR TITLE
[ENG-28888] fix: template validation on create new journey

### PR DESCRIPTION
**Pull request by matheus-sunderhus**
Certainly! Here are the summaries for each product change:

### Product Change: ListSolutions Component
- Added alt text to enhance accessibility for the `ListSolutions` component.

### Product Change: Refactoring create-modal-block
- Improved code readability by using the route name for redirection from `create-modal-block`.
- Refactored code, switching from options API to composition API in the `create-modal` component.
- Simplified the conditional render logic in `create-modal-block`.
- Fixed conditional render to display the template section in `create-modal-block`.

### Product Change: IntegrationInstall View
- Renamed a variable to provide more context within the `IntegrationInstall` view.

### Product Change: LoadingListTemplate Component
- Removed clickable skeleton from the `LoadingListTemplate` component.
- Eliminated a duplicated `div` tag in the `LoadingListTemplate` component.

### Product Change: Codebase Cleanup
- Eliminated unnecessary comments from the codebase.

### Product Change: Schema Creation for Template Engine Block
- Ensured the creation and validation of the text field in the schema creation of `T`.

### Product Change: Password Field Updates and Schema Validation
- Ensured the updating of password field values and ran schema validation.

### Product Change: TemplateEngineBlock
- Removed an unnecessary import from the `TemplateEngineBlock`.

### Product Change: Vendor-related Functionality
- Ensured the call to `loadSolutionByVendor` only when a valid vendor is present.

🎥

https://github.com/aziontech/azion-console-kit/assets/101186721/979f9be2-acb0-4e36-bb98-2d877b143c22

### CODE REVIEW

File: src/templates/create-modal-block/index.vue [hunk 1]
```diff
@@ -1,3 +1,162 @@
...
+  const loadRecommendedSolutions = async () => {
+    try {
+      isLoading.value = true
+      const payload = { type: recommendedHeader.value }
+      templates.value = await MarketplaceService.listSolutionsService(payload)
+    } catch (error) {
+      toast.add({
+        closable: true,
+        severity: 'error',
+        summary: error
```
Comment (HerbertJulio) [inline src/templates/create-modal-block/index.vue:123]: @matheus-sunderhus não seria melhor atualizar o toast, colocando o texto na prop "detail" ?

File: src/templates/create-modal-block/index.vue [hunk 2]
```diff
@@ -1,3 +1,162 @@
...
+    try {
+      isLoading.value = true
+      const payload = { type: browseHeader.value }
+      browseTemplates.value = await MarketplaceService.listSolutionsService(payload)
+    } catch (error) {
+      toast.add({
+        closable: true,
+        severity: 'error',
+        summary: error
+      })
```
Comment (HerbertJulio) [inline src/templates/create-modal-block/index.vue:140]: @matheus-sunderhus não seria melhor atualizar o toast, colocando o texto na prop "detail" ?

File: src/templates/create-modal-block/index.vue [hunk 3]
```diff
@@ -1,3 +1,162 @@
...
+    } finally {
+      isLoading.value = false
+    }
+  }
+
+  const onMenuChange = async (target) => {
+    if (target.value === null) {
```
Comment (HerbertJulio) [inline src/templates/create-modal-block/index.vue:147]: Esse if não poderia ser algo como : 
```suggestion
    if (!target.value) {
```

File: src/templates/create-modal-block/index.vue [hunk 4]
```diff
@@ -1,3 +1,162 @@
...
+      selectedTabControl.value = selectedTab.value
+    } else {
+      selectedTab.value = target.value
+    }
+    if (target.value === 'browse') {
+      if (browseTemplates.value.length === 0) {
```
Comment (HerbertJulio) [inline src/templates/create-modal-block/index.vue:153]: Sugestão: talvez poderia colocar algo como:
```suggestion
      if (!browseTemplates.value.length) {
```

**Review by HerbertJulio — APPROVED** [review]
Fiz algumas sugestões, porém elas não são impedimentos.

File: src/templates/create-modal-block/index.vue (hunk 1 above)
Comment (matheus-sunderhus) [reply src/templates/create-modal-block/index.vue:123]: boa, acabou pegando esse trecho porque devo ter ajustado o escopo superior a isso e não me atentei. Vou  ajustar aqui

File: src/templates/create-modal-block/index.vue (hunk 2 above)
Comment (matheus-sunderhus) [reply src/templates/create-modal-block/index.vue:140]: boa!

File: src/templates/create-modal-block/index.vue (hunk 3 above)
Comment (matheus-sunderhus) [reply src/templates/create-modal-block/index.vue]: boa, refatorei o método também

**Review by HerbertJulio — APPROVED** [review]

**Review by csfeijo — APPROVED** [review]

**Pull request merged**